src/plugins/intel_cpu/src/nodes/executors/acl/acl_fullyconnected.cpp

```diff
@@ -242,26 +315,33 @@ arm_compute::Status ACLFullyConnectedExecutor::validateTensorsInfo(const ACLInfo
     if (aclfcAttrs.isConvertedWeights) {
         aclMemoryInfos[ACLArgs::ACL_WEI]->set_data_type(aclMemoryInfos[ACLArgs::ACL_SRC_0]->data_type());
     }
+    int ic_total = aclMemoryInfos[ACLArgs::ACL_SRC_0]->dimension(0);
     return arm_compute::NEFullyConnectedLayer::validate(
             aclMemoryInfos[ACLArgs::ACL_SRC_0].get(),
-            aclMemoryInfos[ACLArgs::ACL_WEI].get(),
+            &wei_tensor_info,
             aclMemoryInfos[ACLArgs::ACL_BIAS].get(),
             aclMemoryInfos[ACLArgs::ACL_DST].get(),
             fullyConnectedLayerInfo,
-            weightsInfo);
+            expectedWeightFormat == arm_compute::WeightFormat::UNSPECIFIED ?
+                                    arm_compute::WeightsInfo() :
+                                    arm_compute::WeightsInfo(false, 1, 1, ic_total, false, expectedWeightFormat));
 }
 
 ACLFunction ACLFullyConnectedExecutor::configureFunction(const ACLTensors & aclMemoryTensors) {
     auto neFC = std::make_unique<arm_compute::NEFullyConnectedLayer>();
+    aclMemoryTensors[ACLArgs::ACL_WEI]->allocator()->init(wei_tensor_info);
+    int ic_total = aclMemoryTensors[ACLArgs::ACL_WEI]->info()->dimension(0);
     neFC->configure(
             aclMemoryTensors[ACLArgs::ACL_SRC_0].get(),
             aclMemoryTensors[ACLArgs::ACL_WEI].get(),
             aclMemoryTensors[ACLArgs::ACL_BIAS].get(),
             aclMemoryTensors[ACLArgs::ACL_DST].get(),
             fullyConnectedLayerInfo,
-            weightsInfo);
+            expectedWeightFormat == arm_compute::WeightFormat::UNSPECIFIED ?
+                                    arm_compute::WeightsInfo() :
+                                    arm_compute::WeightsInfo(false, 1, 1, ic_total, false, expectedWeightFormat));
 
-    if (aclfcAttrs.isConvertedWeights || !aclfcAttrs.weightsNonTransposed) {
+    if (aclfcAttrs.isWeightsRepacked || aclfcAttrs.isConvertedWeights) {
         aclTensorAttrs.memoryUsageIndicator[ACLArgs::ACL_WEI] = false;
```

Comment:
Let's get rid of those flags and decide whether to import memory or not just based on input type.
Can be done in a follow up PR.

Reply:
Comment is added